src/Form/Type/TpayRetryPaymentDetailsType.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CommerceWeavers\SyliusTpayPlugin\Form\Type;
+
+use CommerceWeavers\SyliusTpayPlugin\BlikPayment\Payum\Factory\GatewayFactory as BlikGatewayFactory;
+use CommerceWeavers\SyliusTpayPlugin\CardPayment\Form\Type\TpayCardType;
+use CommerceWeavers\SyliusTpayPlugin\CardPayment\Form\Type\TpayCreditCardChoiceType;
+use CommerceWeavers\SyliusTpayPlugin\GooglePayPayment\Payum\Factory\GatewayFactory as GooglePayGatewayFactory;
+use CommerceWeavers\SyliusTpayPlugin\GooglePayPayment\Validator\Constraint\EncodedGooglePayToken;
+use CommerceWeavers\SyliusTpayPlugin\PayByLinkPayment\Payum\Factory\GatewayFactory as PBLGatewayFactory;
+use CommerceWeavers\SyliusTpayPlugin\PayByLinkPayment\Validator\Constraint\ValidTpayChannel;
+use CommerceWeavers\SyliusTpayPlugin\VisaMobilePayment\Payum\Factory\GatewayFactory as VisaMobileGatewayFactory;
+use Sylius\Component\Core\Model\PaymentMethodInterface;
+use Sylius\Component\Core\Model\ShopUserInterface;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\CheckboxType;
+use Symfony\Component\Form\Extension\Core\Type\HiddenType;
+use Symfony\Component\Form\Extension\Core\Type\TelType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Validator\Constraints\Length;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\Constraints\Regex;
+
+final class TpayRetryPaymentDetailsType extends AbstractType
+{
+    public function __construct(private readonly TokenStorageInterface $tokenStorage)
+    {
+    }
+
+    /**
+     * @param array<string, mixed> $options
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder
+            ->add(
+                'card',
+                TpayCardType::class,
+                [
+                    'property_path' => '[card]',
+                ],
+            )
+            ->add(
+                'blik_token',
+                TextType::class,
+                [
+                    'data' => null,
+                    'property_path' => '[blik_token]',
+                    'label' => 'commerce_weavers_sylius_tpay.shop.order_summary.blik.token',
+                    'validation_groups' => function (FormInterface $form): array {
+                        return [$this->getValidationGroup($form)];
+                    },
+                    'constraints' => [
+                        new NotBlank(groups: ['payment_retry_' . BlikGatewayFactory::NAME]),
```

Comment:
we should add some prefix to custom validation group

```suggestion
                        new NotBlank(groups: ['cw_payment_retry_' . BlikGatewayFactory::NAME]),
```